OpenRVS/classes/OpenMultiPlayerWidget.uc

```diff
@@ -308,22 +329,219 @@ function Refresh(bool bActivatedByUser)
 	if ( ( m_LanServers.m_ClientBeacon == none ) || ( !m_LanServers.m_ClientBeacon.IsA('OpenClientBeaconReceiver') ) )
 		return;
 
+	//1.5 - if initiated by user, clear the list of current queries and start fresh
+	if ( bActivatedByUser )
+		OpenQueries.remove(0,OpenQueries.length);
 	//0.9: get each server in the list, then query for more info
+	//1.5 - query ServerList instead
+	j = 0;
+	while ( j < ServerList.length )
+	{
+		i = 0;
+		bFound = false;
+		//1.5 - only one open query to a server at a time
+		while ( i < OpenQueries.length )
+		{
+			if ( OpenQueries[i] == ServerList[j].IP )//aleady have query open
+			{
+				bFound = true;
+				break;
+			}
+			i++;
+		}
+		if ( !bFound )
+		{
+			OpenQueries[OpenQueries.length] = ServerList[j].IP;
+			//1.5 - get rough ping
+			if ( Timer == none )
+			{
+				Timer = new class'OpenTimer';
+				Timer.ClockSource = GetLevel();
+			}
+			Timer.StartTimer(ServerList[j].IP);
+			OpenClientBeaconReceiver(m_GameService.m_ClientBeacon).QuerySingleServer(self,
+				Left(ServerList[j].IP,InStr(ServerList[j].IP,":")),
+				int(Mid(ServerList[j].IP,InStr(ServerList[j].IP,":")+1))+1000);
+		}
+		j++;
+		//CurServer = R6WindowListServerItem(CurServer.Next);
+	}
+}
+
+//0.8 written
+//1.3 added mod keyword locking
+//1.5 receive locked info from specific servers, not master list
+function ReceiveServerInfo(string sIP,coerce int iNumP,coerce int iMaxP,string sGMode,string sMapName,string sSvrName,string sModName,bool bSvrLocked)
+{
+	local R6WindowListServerItem CurServer;
+	local int i,iTime;
+/*

```

Comment:
The code using the R6WindowListServerItem has been commented out for now and we're directly iterating through the ServerList array.  This is done to remove dependency on menu list items.  We should have menu list items only display the underlying info in ServerList, not actually be used for storing important data.
This is important as well because it was the fact that we used list items to hold the received server info, that made that info disappear when GetGSServers() got called.